dropwizard-jooq/src/main/java/com/bendb/dropwizard/jooq/jersey/DataSourceParam.java

```diff
@@ -0,0 +1,10 @@
+package com.bendb.dropwizard.jooq.jersey;
+
+import java.lang.annotation.*;
+
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.PARAMETER})
+public @interface DataSourceParam {

```

Comment:
Jersey's integration with HK2 doesn't have a nice implementation to handle @Named nor qualifiers when working with @Context. I've implemented a ValueFactoryProvider to do this instead.